### ISSUE TITLE
`gr.HighlightedText()` problems when `entities` not sorted by start character

### ISSUE THREAD

**Issue by oconnoob** [gradio-app/gradio#2217]
### Describe the bug

# Problem (N.B. Temporary solution below)
If the `entities` of a `gr.HighlightedText()` component are not sorted by start, you get a repeating text issue

```python
d = {
  'text': 'Four score and seven years ago our fathers brought forth on this continent a new nation conceived in liberty and dedicated to the proposition that all men are created equal.\n\nFour score and seven years ago our fathers brought forth on this continent a new nation conceived in liberty and dedicated to the proposition that all men are created equal.',
  'entities': [
    {
      'entity': 1,
      'start': 101,
      'end': 108
    },
    {
      'entity': 1,
      'start': 276,
      'end': 283
    },
    {
      'entity': 1,
      'start': 77,
      'end': 87
    },
    {
      'entity': 1,
      'start': 252,
      'end': 262
    }
  ]
}

with gr.Blocks(css="body {background-color: red}") as demo:
    gr.HighlightedText(d)
    
demo.launch()
```
# Output:
![image](https://user-images.githubusercontent.com/46252299/189216487-c0e372bb-9875-488e-86c4-77b02ea939d9.png)

# Temporary Solution
Apparently if the starting characters of the entities are out of order, this issue will happen. Upon rearranging the elements of the `'entities'` list using `d['entities'] = sorted(d['entities'], key= lambda x: x['start'])` I get the appropriate output.

# Remaining Issues - Line Breaks
In addition, the line breaks present in the text are not rendered by the `HighlightedText()` component.

Thanks!

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

```python
d = {
  'text': 'Four score and seven years ago our fathers brought forth on this continent a new nation conceived in liberty and dedicated to the proposition that all men are created equal.\n\nFour score and seven years ago our fathers brought forth on this continent a new nation conceived in liberty and dedicated to the proposition that all men are created equal.',
  'entities': [
    {
      'entity': 1,
      'start': 101,
      'end': 108
    },
    {
      'entity': 1,
      'start': 276,
      'end': 283
    },
    {
      'entity': 1,
      'start': 77,
      'end': 87
    },
    {
      'entity': 1,
      'start': 252,
      'end': 262
    }
  ]
}

with gr.Blocks(css="body {background-color: red}") as demo:
    gr.HighlightedText(d)
    
demo.launch()
```

### Logs

```shell
N/A
```

### System Info

```shell
Gradio 3.2, Edge/Chrome, Win11
```

### Severity

annoying

**Comment by abidlabs** [comment]
Thanks @oconnoob for creating the issue and providing the example to reproduce it. I can confirm it on my end as well. We'll get a fix in soon!

**Comment by abidlabs** [comment]
Fixed by #2221, thanks @oconnoob!
